### ISSUE TITLE
maxstd and minstd are in conflict

### ISSUE THREAD

**Issue by nikohansen** [CMA-ES/pycma#49]
In particular if these values are set close to each other, they cannot necessarily be satisfied at the same time, because the global `sigma` is used for both corrections. This could be easily fixed in the `CMA_diagonal` setting, where diagonal corrections are unproblematic. With full covariance matrix this is rather a feature than a bug, because the change size of a diagonal change is non-trivial to compute. Alternatively to adjust `maxstd`, the axis/axes with the largest projection could be shortened.

As a side note, decreasing `sigma` to set `maxstd` seems only a good idea if it corrects for a previous increase of `sigma`.

**Comment by nikohansen** [comment]
The current code
```python
        if any(self.sigma * self.sigma_vec.scaling * self.dC**0.5 <
                       np.asarray(self.opts['minstd'])):
            self.sigma = max(np.asarray(self.opts['minstd']) /
                                (self.sigma_vec * self.dC**0.5))
            assert all(self.sigma * self.sigma_vec * self.dC**0.5 >=
                       (1-1e-9) * np.asarray(self.opts['minstd']))
        elif any(self.sigma * self.sigma_vec.scaling * self.dC**0.5 >
                       np.asarray(self.opts['maxstd'])):
            self.sigma = min(np.asarray(self.opts['maxstd']) /
                             self.sigma_vec * self.dC**0.5)
```
seems wrong for setting `'maxstd'` in the last `elif`. It also uses an outdated interface.

Given we have `self.sigma_vec` available, it seems a good choice to handle min and max component-wise.

**Comment by nikohansen** [comment]
Fixed with commit 33c0573853e26e8bb583b478dd178b265cf3d733
